### ISSUE TITLE
add support (maybe via plugin to watch for file changes)

### ISSUE THREAD

**Issue by gaborbernat** [tox-dev/tox#603]
some tools offer this possibility, we should provide a way to do this with tox also (not necessarily part of tox); something similar to https://github.com/joeyespo/pytest-watch

**Comment by gaborbernat** [comment]
I think the main challenge with this is that this would imply having the ability to specify files that needs to be watched. Maybe we can by default do it for all python files from ``toxinidir`` and give the options so  that the user can override it via an expression; e.g. ``tests/**/*.py``. In the meantime as a workaround people can use the environment tools available; e.g. for running ``pytest`` add ``pytest-watch`` and then start tox by passing the invocation flag; e.g. ``tox -e py27 -- --watch``.  It's up to debate if this should be done altogether at tool level (e.g. flake8; pytest) as the tools already know what files to watch.  What do you guys think? Is this the invoked commands responsibility or not?

**Comment by obestwalter** [comment]
Which files to watch is actually the hard part and I think needs to be configurable. Watching toxinidir might be a bit much in the default config, as this would include all the cruft that usually piles up in a developer working copy including the `.tox` folder.

I am all for wrapping existing functionality instead of reinventing the whhel, so using e.g. pytest-watch would not be a workaround for me, but part of the solution.

**Comment by gaborbernat** [comment]
yeah, you're right the ``.tox`` definitely needs to be ignored. My main worry with adding this part of tox is configuration duplication between tools builtin support (e.g. ``pytest-watch``, or ``pytest-xdist -f```) and tox file watch support.
